Compiling error. (OpenWRT ar71xx MIPSBE)

Hi, looks like latest commits caused some issues when compiling for some systems.

I'm compiling for the WiFi Pineapples. (ar71xx CPU / MIPS-BE).
I noticed some updates lately regarding endianess, that's why i'm referencing what system i'm compiling for.

Here's the error message:
```
mips-openwrt-linux-uclibc-gcc -Os -pipe -mno-branch-likely -mips32r2 -mtune=34kc -fno-caller-saves -fhonour-copts -Wno-error=unused-but-set-variable -msoft-float -mips16 -minterlink-mips16  -MMD -MF .deps/hcxpcaptool.d -o hcxpcaptool hcxpcaptool.c -lz -lcrypto 
hcxpcaptool.c: In function 'processpcapng':
hcxpcaptool.c:5684:2: error: expected ';' before 'if'
  if(blocktype == PCAPNGBLOCKTYPE)
  ^
make[3]: *** [Makefile:79: hcxpcaptool] Error 1
```

--- a/hcxpcaptool.c
+++ b/hcxpcaptool.c
@@ -5679,7 +5679,7 @@ while(1)
 	#ifdef BIG_ENDIAN_HOST
 	blocktype  = byte_swap_32(blocktype);
 	blocklen = byte_swap_32(blocklen);
-	blockmagic = byte_swap_32(blockmagic)
+	blockmagic = byte_swap_32(blockmagic);
 	#endif
 	if(blocktype == PCAPNGBLOCKTYPE)
 		{

